File: examples/langchain-js/veritas-research-tool.ts
import { tool } from "langchain";
import * as z from "zod";
import { ResearchClient } from "@veritas/sdk";

const client = new ResearchClient({
  apiKey: process.env.VERITAS_API_KEY!,
  baseUrl: process.env.VERITAS_API_URL || "http://localhost:3000/v1",
});

export const veritasResearchTool = tool(
  async ({ query, mode, sessionId }) => {
    const result = await client.research({
      query,
      mode,
      sessionId: sessionId || "langchain-default",
      outputSchema: {
        type: "object",
        properties: {
          summary: { type: "string" },
          key_findings: { type: "array", items: { type: "string" } },
          sources: { type: "array" },
        },
      },
    });

    return JSON.stringify({
      status: result.status,
      confidence_score: result.confidenceScore,
      data: result.data,
      sources: result.sources,
      error: result.error,
    });
  },
  {
    name: "veritas_research",
    description:
      "Run trusted multi-source web research. Use this when an agent needs current facts, source checks, or structured research output.",
    schema: z.object({
      query: z.string().describe("Research question to investigate."),
      mode: z.enum(["lite", "medium", "deep"]).default("medium"),
      sessionId: z.string().optional().describe("Stable session id for follow-up research."),
    }),
  }
);

File: examples/langchain-python/veritas_research_tool.py
import json
import os

from langchain.tools import tool
from researchplatform import ResearchClient, ResearchMode


client = ResearchClient(
    api_key=os.environ["VERITAS_API_KEY"],
    base_url=os.getenv("VERITAS_API_URL", "http://localhost:3000/v1"),
)


@tool("veritas_research")
def veritas_research(query: str, mode: str = "medium", session_id: str = "langchain-default") -> str:
    """Run trusted multi-source web research for current facts, source checks, or structured research output."""
    result = client.research(
        query=query,
        mode=ResearchMode(mode),
        session_id=session_id,
        output_schema={
            "type": "object",
            "properties": {
                "summary": {"type": "string"},
                "key_findings": {"type": "array", "items": {"type": "string"}},
                "sources": {"type": "array"},
            },
        },
    )
    return json.dumps(
        {
            "status": result.status,
            "confidence_score": result.confidence_score,
            "data": result.data,
            "sources": result.sources,
            "error": result.error,
        }
    )
